File: eecs268/2021spring/MWF/2021.03.05.inheritance-part-3/Animal/Dog.cpp
//Dog.cpp

#include "Dog.h"

                   //initializer list
Dog::Dog(int age) : Animal(age)
{
	std::cout << "Dog constructed!\n";
	m_isHungry = false;
}

void Dog::doTrick()
{
	std::cout << "Dog is rolling over\n";
}

void Dog::sleep()
{
	std::cout << "Rup. Rup. Rup-up\n";
}

File: eecs268/2025fall/MWF/2025.12.01.inheritance/inheritance/dog.py
#dog.py

from animal import Animal

class Dog(Animal):
    def __init__(self, age, tail_length):
        super().__init__(age)
        print('Dog created')
        self.tail_length = tail_length
        
    def do_trick(self):
        print('This', self.age, 'year old dog')
        print('Dog doing an amazing trick!')

    def sleep(self):
        self.do_trick()
        print('Hrp. Hrp. Hrp HRP.')
